pyiron_base/job/template.py

```diff
@@ -23,29 +24,62 @@
 class TemplateJob(GenericJob):
     def __init__(self, project, job_name):
         super().__init__(project, job_name)
-        self.input = GenericParameters(table_name="input")
+        self._input = InputList(table_name="input")
```

Comment:
I like this. Also the associated infrastructure of the property and setter, and making sure this field is automatically taken care of in `to_hdf` and `from_hdf`. I'm similarly happy to see output treated this way. 

What I would really like to see is for all of this to be taking place in `GenericJob`. Other than the fact we'd need to do the leg work of updating all the downstream classes, is there a reason not to do this? I feel `GenericJob` should by default have `input` and `output` fields, and this is the most up-to-date way of handling these I've seen.

Reply:
Yes, a generic way of handling input and output would be great! Furthermore, I am not even sure things would break immediately if these objects are given by default on the `GenericJob level`?

Reply:
Do you have all the repos and test environments nicely set up on your machine? I would suggest simply trying pushing the change upstream to `GenericJob`, and running the tests to see whether you're right about that!

It's 100% a direction I want to go in, but if it's going to break things then I understand if we don't go there in this PR.

Reply:
I will test if my test suite is up to date and try to break thinks! However, I am not sure how rigorous the tests are checking what is actually written to the hdf file. With the table_name="input" we would write stuff to the same group, where we usually write to. This might play well, if the fields are orthogonal, but does not have to.

Reply:
Just a quick update at this point: Tests are failing at least for the output field if I just shift these changes upstream. I could of course rename this like io_output and io_input to avoid name clashes with usually used fields.
Furthermore, I have to update and enhance my test environment. The unittests will work at some point, however, I probably have to run the notebooks on the cluster - for some reason the Laptop runs out of memory - maybe due to the windows subsystem setup....